File: Assignment16/Program1.java
/*Write a program to calculate the sum of first N natural numbers.*/
package Assignment16;
class Logic {
    private int n;

    public Logic(int n)
    {
        this.n = n;
    }
    public void calculateSum() 
    {
        int iSum = 0;
        for (int iCnt = 1; iCnt <= this.n; iCnt++) 
        {
            iSum = iSum + iCnt;
        }
        
        System.out.println("The sum of numbers from 1 to " + this.n + " is: " + iSum);
    }
}
class Program1
{
    public static void main (String A[])
    {
        Logic obj = new Logic(10);
        obj.calculateSum();
    }
}

File: Assignment43/program1.c
/*
1.  Write a program which displays all elements which are perfect from singly linear list.
*/
#include <stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;


void InserFirst(PPNODE first, int no)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));
    
    newn->data = no;
    newn->next = NULL;

    if(*first == NULL)  //LL is empty
    {
        (*first) = newn;
    }
    else                //LL contains atleast 1 node
    {
        newn->next = (*first);
        (*first) = newn;
    }
}

void Display(PNODE first)
{
    while(first!=NULL)
    {
        printf("| %d |->", first->data);
        first= first ->next;
    }
    printf(" NULL\n");
}

void DisplayPerfect(PNODE Head)
{
    int iCnt = 0;
    int iSum = 0;
    int iNo = 0;

    printf("Perfect numbers in the list are: \n");

    while(Head != NULL)
    {
        iNo = Head->data; 
        iSum = 0;         

        for(iCnt = 1; iCnt <= (iNo / 2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                iSum = iSum + iCnt;
            }
        }

        if(iSum == iNo)
        {
            printf("%d\t", iNo);
        }
        Head = Head->next;
    }
    printf("\n");
}
int main()
{
    PNODE head = NULL;
    int iRet = 0;

    InserFirst(&head, 6);
    InserFirst(&head, 89);
    InserFirst(&head, 41);
    InserFirst(&head, 17);
    InserFirst(&head, 28);
    InserFirst(&head, 11);

    Display(head);

    DisplayPerfect(head);

    return 0;
}